src/main/java/net/yogstation/yogbot/bans/StickyRoleManager.kt

```diff
@@ -28,63 +30,39 @@ class StickyRoleManager(
 		if (member == null) return logChannel.log("Unable to check for sticky roles")
-		var stickyRoleCount = 0
-		try {
-			databaseManager.yogbotDbConnection.use { connection ->
-				connection.prepareStatement("INSERT IGNORE INTO user_sticky_roles (discord_id, role_id) VALUES (?, ?)")
-					.use { statement ->
-						member.roleIds.filter { discordConfig.isStickyRole(it.asLong()) }.forEach {
-							stickyRoleCount++
-							statement.setLong(1, member.id.asLong())
-							statement.setLong(2, it.asLong())
-							statement.addBatch()
-						}
-						if (stickyRoleCount > 0) statement.executeBatch()
-					}
+		return Mono.fromCallable {
+			stickyRoleRepository.saveAll(member.roleIds.filter { discordConfig.isStickyRole(it.asLong()) }
+				.map { StickyRole(member.id.asLong(), it.asLong()) }).count()
+		}.subscribeOn(Schedulers.boundedElastic()).flatMap { count ->
+			if (count > 0) {
+				member.privateChannel.flatMap { channel -> channel.createMessage("You have left the ${discordConfig.serverName} server with $count sticky role${if (count != 1) "s" else ""}. They will be reapplied when you rejoin.") }
+					.and(logChannel.log("**${member.tag}** left the server with $count sticky role${if (count != 1) "s" else ""}"))
+			} else {
+				Mono.empty<Any>()
 			}
-		} catch (e: SQLException) {
-			logger.error("Error saving sticky roles", e)
-			return logChannel.log("A SQL Error occurred while saving sticky roles")
 		}
-		if (stickyRoleCount > 0) {
-			return member.privateChannel.flatMap { channel -> channel.createMessage("You have left the ${discordConfig.serverName} server with $stickyRoleCount sticky role${if (stickyRoleCount != 1) "s" else ""}. They will be reapplied when you rejoin.") }
-				.and(logChannel.log("**${member.tag}** left the server with $stickyRoleCount sticky role${if (stickyRoleCount != 1) "s" else ""}"))
-		}
-		return Mono.empty<Any>()
 	}
 
 	/**
 	 * Checks if a user logged off with sticky bans, if they did, reapplies them
 	 * @param member The member to check
 	 */
 	fun doLogin(member: Member): Mono<*> {
-		var stickyRoleCount = 0
-		var resultMono: Mono<*> = Mono.empty<Any>()
-		try {
-			databaseManager.yogbotDbConnection.use { connection ->
-				connection.prepareStatement("SELECT role_id FROM user_sticky_roles WHERE discord_id = ?")
-					.use { statement ->
-						statement.setLong(1, member.id.asLong())
-						statement.executeQuery().use { resultSet ->
-							while (resultSet.next()) {
-								stickyRoleCount++;
-								resultMono = resultMono.and(member.addRole(Snowflake.of(resultSet.getLong("role_id"))))
-							}
-						}
-					}
-				connection.prepareStatement("DELETE FROM user_sticky_roles WHERE discord_id = ?").use { statement ->
-					statement.setLong(1, member.id.asLong())
-					statement.executeUpdate()
-				}
+		return Mono.fromCallable {
+			val roles: List<StickyRole> = stickyRoleRepository.findAllByDiscordId(member.id.asLong()).filter { discordConfig.isStickyRole(it.roleId) }

```

Comment:
I would probably try and move a lot of the filtering and such behavior into the repositories, personal preference though.

Reply:
The filter is there because its not a query filter, but rather a filter on the returned data